### ISSUE TITLE
type parameter of layout function is __untyped__

### ISSUE THREAD

**Issue by agdev** [yatskevich/grunt-bower-task#121]
Hi,

When I run grunt bower w/ following config I see that type'd value is "**untyped**"
Why is that?
bower: {
            install: {
                options: {
                    targetDir: './components',
                    install: true,
                    layout: function (type, component) {
                        grunt.log.writeln('type="' + type + '"');
                        grunt.log.writeln('component="' + component + '"');
                        var renamedType = type;
                        if (type == 'js') renamedType = 'scripts';
                        else if (type == 'css') renamedType = 'styles';

```
                    return path.join(component, renamedType);
                },
                bowerOptions: {forceLatest:true}
            }
        }
    }
```

Thank you.

**Comment by wpoch** [comment]
Did you get any workaround?, Is happening to me too. I want to integrate bower with sailsJs

**Comment by agdev** [comment]
No, I did not, was planning to write my own grunt plugin but did not get to it.

**Comment by mrchief** [comment]
A workaround is to inspect the 3rd argument passed and determine type based on file extension:

```
layout: function (type, component, source) {
    /* workaround for https://github.com/yatskevich/grunt-bower-task/issues/121 */
    if (type === '__untyped__') {
        type = source.substring(source.lastIndexOf('.') + 1);
    }

    var renamedType = type;

    if (type === 'js') renamedType = './Scripts';
    else if (type === 'css') renamedType = './Content';

    return path.join(component, renamedType);
}
```

**Note**: You can also use `type = path.extname(source).replace('.', '')` instead of `type = source.substring(source.lastIndexOf('.') + 1);`
